Common/Chairloader/ConfigNode.h

```diff
@@ -0,0 +1,247 @@
+    }
+
+    ConfigNode create(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            throw std::runtime_error("Node already exists");
+        }
+        auto node = m_node.append_child(name);
+        node.append_attribute("type") = toString(type).c_str();
+        node.append_child(pugi::node_pcdata).set_value("");
+        return ConfigNode(node);
+    }
+
+    ConfigNode getOrCreate(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            return ConfigNode(m_node.child(name));
+        }
+        return create(name, type);
+    }
+
+    bool exists() const {
+        return m_node;
+    }
+
+    NodeType type() const {
+        return m_nodeType;
+    }
+
+    std::string name() const {
+        return m_name;
+    }
+
+    std::string displayName() const {
+        return m_displayName;
+    }
+
+    std::string description() const {
+        return m_description;
+    }
+
+    pugi::xml_node raw() const {
+        return m_node;
+    }
+
+    template<typename T>
+    T as() const {
+        if(!m_node) {
+            throw std::runtime_error("Node does not exist");
+        }
+        if(m_nodeType == NodeType::Enum){
+            return ConfigNode(m_node.child("selected")).as<T>();
+        } else {
+            return getValue<T>(m_node);
+        }
+    }
+
+    /**
+     * Get the value of the node with a default value if the node is not found.
+     * @return The value of the node or the default value if the node does not exist or is not of the expected type.
+     */
+    template<typename T>
+    T asOr(T defaultValue) const noexcept{
+        try {
+            if(m_nodeType == NodeType::Enum){
+                return ConfigNode(m_node.child("selected")).asOr<T>(defaultValue);
+            }
+            return exists() ? getValue<T>(m_node) : defaultValue;
+        } catch (...) {
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Get the value of the node as an optional. If the node does not exist, an empty optional is returned.
+     * @return The value of the node as an optional.
+     */
+    template<typename T>
+    std::optional<T> maybe() const noexcept {
+        if(!exists()) return std::nullopt;
+        try {
+            if(m_nodeType == NodeType::Enum){
+                return ConfigNode(m_node.child("selected")).maybe<T>();
+            } else {
+                return getValue<T>(m_node);
+            }
+        } catch (...) {
+            return std::nullopt;
+        }
+    }
+
+    /**
+     * Set the value of the node. If the node does not exist, an exception is thrown.
+     *
+     * Note: does not actually save the value to the XML file, just sets the value in memory.
+     * @param value The value to set.
+     */
+    template<typename T>
+    void set(const T& value, NodeType type = NodeType::Unknown) {
+        if(!m_node) {
+            throw std::runtime_error("Node does not exist");
+        }
```

Comment:
This particular check may be moved into a private method `CheckNodeValid()` because it is used in many places